packages/time-series-chart/src/Chart.tsx

```diff
@@ -0,0 +1,223 @@
+const FIRST_COLUMN_INVALID_MSG = "Wrong type for first column, it should be either LABEL, TEXT, NUMBER or DATE.";
+const NOT_ENOUGH_COLUMNS_MSG_1 =
+  "Time series component expects 3 columns: category(LABEL or TEXT or NUMBER or DATE), series(TEXT) and values(NUMBER).";
+const FIRST_COLUMN_INVALID_MSG_1 = "Wrong type for first column, it should be either LABEL, TEXT, NUMBER or DATE.";
+const SECOND_COLUMN_INVALID_MSG_1 = "Wrong type for second column, it should be TEXT.";
+const THIRD_COLUMN_INVALID_MSG_1 = "Wrong type for third column, it should be NUMBER.";
+
+enum Params {
+  TRANSPOSED = "transposed",
+  CHARTNAME = "chartName"
+}
+
+enum AppStateType {
+  ERROR = "Error",
+  INIT = "Initializing",
+  LOADING_COMPONENT = "Loading Component",
+  LOADED_COMPONENT = "Loaded Component",
+  FINISHED = "Finished loading"
+}
+
+interface AppState {
+  state: AppStateType;
+  processesoptions: Options;
+  processesseries: SingleSeries[];
+  configurationIssue: string;
+  message?: string;
+}
+
+const validateParams = (params: Map<string, string | number>): string | undefined => {
+  if (!params.get(Params.TRANSPOSED)) {
+    return "Transposed is required.";
+  }
+  if (!params.get(Params.CHARTNAME)) {
+    return "Chart name is required.";
+  }
+};
+
+const validateDataSet = (ds: DataSet, transposed: boolean) => {
+  if (transposed) {
+    validateTransposedDataset(ds);
+  } else {
+    validateNonTransposedDataset(ds);
+  }
+};
+
+const validateNonTransposedDataset = (ds: DataSet): string | undefined => {
+  if (ds.columns.length < 2) {
+    return NOT_ENOUGH_COLUMNS_MSG;
+  }
+  for (let i = 1; i < ds.columns.length; i++) {
+    if (ds.columns[i].type !== ColumnType.NUMBER) {
+      return "Wrong type for column" + (i + 1) + ", it should be NUMBER";
+    }
+  }
+};
+
+const validateTransposedDataset = (ds: DataSet): string | undefined => {
+  if (ds.columns.length < 3) {
+    return NOT_ENOUGH_COLUMNS_MSG_1;
+  }
+  if (ds.columns[1].type !== ColumnType.TEXT && ds.columns[1].type !== ColumnType.LABEL) {
+    return SECOND_COLUMN_INVALID_MSG_1;
+  }
+  if (ds.columns[2].type !== ColumnType.NUMBER) {
+    return THIRD_COLUMN_INVALID_MSG_1;
+  }
+};
+
+interface Props {
+  controller: ComponentController;
+}
+
+function getSeries(dataset: DataSet, transposed: boolean): SingleSeries[] {
+  let arrayseries: SingleSeries[] = [];
+  if (transposed) {
+    arrayseries = getSeriesforTransposedDataset(dataset);
+  } else {
+    arrayseries = getSeriesforNonTransposedDataset(dataset);
+  }
+  return arrayseries;
+}
+
+function getSeriesforNonTransposedDataset(dataset: DataSet): SingleSeries[] {
+  const arrayseries: SingleSeries[] = [];
+  let newseries: SingleSeries = { name: "", data: [] };
+  for (let i = 1; i < dataset.columns.length; i++) {
+    newseries = { name: dataset.columns[i].name, data: dataset.data.map((d: Array<string | number>) => d[i]) };
```

Comment:
Would it be possible to use `dataset.columns[i].settings.columnName` instead `columns[i].name` - this way the component will have the column that user configured in the UI instead the column id.